src/test/lib/sql/sql_pipeline_statement_test.cpp

```diff
@@ -96,6 +97,7 @@ class SQLPipelineStatementTest : public BaseTest {
       "> 1000";
   const std::string _multi_statement_query = "INSERT INTO table_a VALUES (11, 11.11); SELECT * FROM table_a";
   const std::string _multi_statement_dependant = "CREATE VIEW foo AS SELECT * FROM table_a; SELECT * FROM foo;";
+  const std::string _parameter_query = "SELECT * FROM table_a WHERE table_a.a > 1000 AND table_a.b < 458";
```

Comment:
We need substantially more test coverage here - not in terms of the percentage of lines covered, but in the variety and complexity of the conditions. The handling of values in subqueries is not tested, neither is the post-cache pruning.

Testing this might be simpler if the query reuse was more encapsulated instead of being grafted on the SQLPipelineStatement.